File: algorithms/1_easy/R/01332_remove_palindromic_subsequences/solution.c
static inline bool is_pal(const char *restrict s) {
    const size_t len = strlen(s);
    ptrdiff_t hi = (ptrdiff_t)len - 1;
    ptrdiff_t lo = 0;
    while (lo < hi) {
        if (s[hi] == s[lo]) {
            lo += 1;
            hi -= 1;
        } else {
            return false;
        }
    }
    return true;
}

int removePalindromeSub(const char *restrict s) {
    if (s == NULL || s[0] == '\0') {
        return -1;
    }
    if (is_pal(s)) {
        return 1;
    }
    return 2;
}

File: algorithms/1_easy/R/01332_remove_palindromic_subsequences/solution.rs
impl Solution {
    #[inline(always)]
    fn is_pal(bytes: &[u8]) -> bool {
        let mut hi = bytes.len() as isize - 1;
        let mut lo = 0_isize;
        while lo < hi {
            if bytes[hi as usize] == bytes[lo as usize] {
                lo += 1;
                hi -= 1;
            } else {
                return false;
            }
        }
        true
    }

    pub fn remove_palindrome_sub(s: String) -> i32 {
        assert!(s.len() != 0);

        if Self::is_pal(s.as_bytes()) {
            return 1;
        }

        2
    }
}
